From a132617c65cc63c45d5851c60bbf9d5eea704f45 Mon Sep 17 00:00:00 2001
From: wwidjaja <w.widjaja@intel.com>
Date: Tue, 25 Feb 2020 14:16:02 +0800
Subject: [PATCH] PONRTSYS-4533 - remove space add EIP123 permission to access
 4kec DDR

  - for safety (secure storage service) add EIP123 as allowed host

  NOTE: any other IA (master) needed ? to be confirmed / test
---
 arch/mips/lantiq/grx500_bootcore/ngi_prx300.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
index 59d2f278e2bb..191ea292b3d0 100644
--- a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
+++ b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
@@ -134,7 +134,7 @@ void set_permission(void)
 	permission_wr(ssx1_base,
 			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_0, ALL_CPUS);
 	#ifdef AUTO_DDR_REGION_PROTECT
-	 permission_wr(ssx1_base,
+	permission_wr(ssx1_base,
 			REG0_TDDR_PM_TDDR_PM_ADDR_MATCH_1, (get_base_ddr_fmt1()|VAL_NGI_DDR_REGION_SIZE));
 	#else /* AUTO_DDR_REGION_PROTECT */
 	/* protect DDR region 0x3E000000~0x40000000 32MB */
@@ -142,9 +142,9 @@ void set_permission(void)
 			REG0_TDDR_PM_TDDR_PM_ADDR_MATCH_1, 0x3E000081);
 	#endif /* AUTO_DDR_REGION_PROTECT */
 	permission_wr(ssx1_base,
-			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_1, M4KEC_ONLY);
+			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_1, M4KEC_ONLY | EIP123);
 	permission_wr(ssx1_base,
-			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_1, M4KEC_ONLY);
+			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_1, M4KEC_ONLY | EIP123);
 	permission_wr(ssx1_base,
 			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_2, ALL_CPUS);
 	permission_wr(ssx1_base,
